scripts/globals/job_utils/thief.lua

```diff
@@ -212,9 +212,17 @@ xi.job_utils.thief.useDespoil = function(player, target, ability, action)
 
     local stolen = target:getDespoilItem()
 
-    if target:isMob() and math.random(100) < despoilChance and stolen then
+    if
+        target:isMob() and
+        math.random(1, 100) <= despoilChance and
+        stolen
+    then
         if player:getObjType() == xi.objType.TRUST then
-            player:getMaster():addItem(stolen)
+            if stolen then
+                player:getMaster():addItem(stolen)
+            else
+                return xi.msg.basic.SKILL_NO_EFFECT
```

Comment:
on retail the trust won't even attempt this if its not going to succeed. it will NEVER give you a no effect message. Aside form that this is only checking the bool `stolen` twice, it being in the original check already should be `false` when the item is nil and thus never reach this point.

Reply:
true... could just remove the else statement then, if i understand correctly? that seems to make sense.

Reply:
I think a better solution would be to add a check for nil despoil item in `onAbilityCheck`

Reply:
> I think a better solution would be to add a check for nil despoil item in `onAbilityCheck`

check if trust and nil, so we don't affect players. the original scritp in lsb that this r originally altered can stay liek we had it and if a nil is every being given to player, rechecking the variable a 2nd time wouldn't have fixed it but would still be an issue to report to us. hopefully the issue was limited to trust despoils anyway.

Reply:
```lua
xi.job_utils.thief.checkDespoil = function(player, target, ability)
    if player:getFreeSlotsCount() == 0 then
        return xi.msg.basic.FULL_INVENTORY, 0
    end

    if player:getObjType() == xi.objType.TRUST then
        if
            player:getMaster():getFreeSlotsCount() == 0 or
            not target::getDespoilItem()
        then
            return 1, 0
        end
    end

    return 0, 0
end
```